ext/standard/var.c

```diff
@@ -71,7 +71,22 @@ static void php_object_property_dump(zval *zv, zend_ulong index, zend_string *ke
 		}
 		ZEND_PUTS("]=>\n");
 	}
-	php_var_dump(zv, level + 2);
+
+	if (Z_TYPE_P(zv) == IS_UNDEF) {
+		if (prop_info->type) {
+			php_printf("%*cuninitialized(%s%s)\n",
+				level + 1, ' ',
+				prop_info->allow_null ? "?" : "",
+				(prop_info->type == IS_OBJECT) ?
+					ZSTR_VAL(prop_info->type_name) :
+					zend_get_type_by_const(prop_info->type));
+		} else {
+			php_printf("%*cuninitialized\n",
+				level + 1, ' ');
+		}
+	} else {
+		php_var_dump(zv, level + 2);
+	}
```

Comment:
I'm not sure if this information about uninitialized properties are useful. It also inconsistent with printed number of properties and behavior of untyped properties.

<?php
class C {
  public $a;
  public int $b;
}
$x = new C;
var_dump($x);
$x->a = $x->b = 5;
var_dump($x);
unset($x->a);
unset($x->b);
var_dump($x);
?>

object(C)#1 (1) {
  ["a"]=>
  NULL
  ["b"]=>
  uninitialized(integer)
}
object(C)#1 (2) {
  ["a"]=>
  int(5)
  ["b"]=>
  int(5)
}
object(C)#1 (0) { // <<-- 
  ["b"]=>
  uninitialized(integer)
}